### PR TITLE
Global url namespace

### PR DESCRIPTION
**Pull request by danpaz**
Addresses #252.  

This is pretty quick & dirty so please lmk if you have any feedback!

* User can add a `namespace` setting to prefix urls, either:
 - On the global mockjax object `$.mockjaxSettings.namespace = '/api/v1'`
 - Or on a particular mock `$.mockjax({url: 'myservice', namespace: '/api/v1'})`.
* Preference is given to the mock-level setting to allow overriding the global setting.
* If no `namespace` is provided everything should work just as before.
* Namespace doesn't apply to proxies.

A couple of the commits are just trimming trailing whitespace because of one of my sublime text settings -- feel free to revert those even just to make the diff easier to read.

### CODE REVIEW

**Comment by jakerella** [comment]
Looks pretty good, thanks for contributing! I'm going to add a few notes inline. Also, can you add a small section to the readme on usage?

Lastly, I think we need to add a couple additional tests:
* Ensure that with a namespace set, a non-matching url does not get mocked
* Ensure that multiple mocks in a row get mocked with a namespace

File: src/jquery.mockjax.js [hunk 1]
```diff
@@ -115,6 +115,16 @@
 				return null;
 			}
 		} else {
+
+			// Apply namespace prefix to the mock handler's url.
+			var namespace = handler.namespace || $.mockjaxSettings.namespace;
+			if (!!namespace) {
+				handler.url = [
+						namespace,
+						handler.url
```
Comment (jakerella) [inline src/jquery.mockjax.js]: Hmm... what if they prefix their handler url with "/"? I think we need to do a tiny bit more on this check to be smart about it.

File: src/jquery.mockjax.js [hunk 2]
```diff
@@ -668,7 +678,7 @@
 
 	var DEFAULT_RESPONSE_TIME = 500;
 
-    
+
 	$.mockjaxSettings = {
```
Comment (jakerella) [inline src/jquery.mockjax.js]: We need to add a default in here, and I think it should probably be `null` (versus the empty string which you have in the readme).

**Comment by danpaz** [comment]
Thanks for the comments, I'll try to tackle it this weekend and will ping you when it's ready for another look!

**Comment by danpaz** [comment]
@jakerella I pushed a few updates. Ready for another look.

**Comment by jakerella** [comment]
Sweet, I'll take a look soon.

File: src/jquery.mockjax.js [hunk 3]
```diff
@@ -115,6 +115,15 @@
 				return null;
 			}
 		} else {
+
+			// Apply namespace prefix to the mock handler's url.
+			var namespace = handler.namespace || $.mockjaxSettings.namespace;
+			if (!!namespace) {
+				var namespacedUrl = [namespace, handler.url].join('/');
+				namespacedUrl = namespacedUrl.replace(/(\/+)/g, "/");
```
Comment (jakerella) [inline src/jquery.mockjax.js]: Can we switch this to single quote style? I know... picky... but our build process runs jshint and _should_ fail on this. :)

File: test/test.js [hunk 4]
```diff
@@ -1980,3 +1980,130 @@ asyncTest('Test for bug #26: jsonp mock fails with remote URL and proxy', functi
...
+
+asyncTest('should be able to override global namespace per-mock', function() {
+  $.mockjaxSettings.namespace = '/api/v1';
+
+  $.mockjax({
+    url: 'myservice',
+    namespace: '/api/v2'
+  });
+
+  $.ajax({
```
Comment (jakerella) [inline test/test.js]: Can we also test that a call to `/api/v1/myservice` fails? I think that's it for the tests.

**Comment by jakerella** [comment]
@danpaz two minor comments, then we should be good to go. FYI, I'm going to hold off until we release `2.0.0` (out of `-beta`) before merging this in. It should be included in the `2.1.0` release which (hopefully) will be soon after.

**Comment by danpaz** [comment]
@jakerella Good comments, thanks! Looking forward to using this feature :)

**Comment by jakerella** [comment]
@danpaz Looks like there are some conflicts in the merge. It's possible that this is just from the `/dist` files, in which case just merge in `master` and then regenerate. In any case, once that's done I'll get this merged.

**Comment by danpaz** [comment]
@jakerella Done and done :bomb:

**Pull request merged**